### PR TITLE
Finetuning Bug Fixes

### PR DESCRIPTION
**Pull request by Satrat**
Various bugfixes for matching up sparseML and llama-recipes finetuning
* removing dummy learning rate scheduler
* fixing data collator
* adding step-wise loss and perplexity logs to mimic llama-recipes
* support for FSDP teacher, and keep teacher in eval mode

### Testing
* llama-recipes runs: https://docs.google.com/spreadsheets/d/1qeEi9ICbu9hkwpnFReOh1jbXShj0dOGYbaFb3M2AVi4/edit#gid=0 
* sparseML runs: https://docs.google.com/spreadsheets/d/1-AMyNT8oY1bDmtkRm3llnhNmK6FMIX7mxYlgcWC4O6c/edit#gid=0

### CODE REVIEW

**Review by bfineran — APPROVED** [review]

**Review by dbogunowicz — APPROVED** [review]
Approved, pending transformers errors in GHA

**Review by bfineran — APPROVED** [review]

**Review by dbogunowicz — APPROVED** [review]
